agent/wpthook/test_state.cc

```diff
@@ -319,7 +325,14 @@ bool TestState::IsDone() {
   if (_active) {
     bool is_page_done = false;
     CString done_reason;
-    if (test_ms >= _test._minimum_duration) {
+    DWORD elapsed_timeout_ms = ElapsedMsFromTimeoutStart(now);
+    if (_test._overall_timeout_in_ms != 0 &&

```

Comment:
Just for safety, can you make it _test._overall_timeout_in_ms > 0. I know you've made sure of it before, but just in case, a bug creeps in, we'd still be safe here.